### ISSUE TITLE
[SDK 51] expo-sensors 13.0.6 crash 

### ISSUE THREAD

**Issue by mlecoq** [expo/expo#28820]
### Minimal reproducible example

https://github.com/mlecoq/expo-sensors-issue

### What platform(s) does this occur on?

Android

### Did you reproduce this issue in a development build?

Yes

### Summary

I have the following error displayed on Android on `DeviceMotion.addListener` call:

```
 Tried to obtain display from a Context not associated with  one. Only visual Contexts (such as Activity or one created with Context#createWindowContext) or ones created with Context#createDisplayContext are associated with displays. Other types of Contexts are typically related to background entities and may return an arbitrary display.
   at android.app.ContextImpl.getDisplay(ContextImpl.java:2580)
```

### Environment

  expo-env-info 1.2.0 environment info:
    System:
      OS: macOS 14.4.1
      Shell: 5.9 - /bin/zsh
    Binaries:
      Node: 20.6.1 - /var/folders/md/1qkynwbs1216d57grkbc682h0000gn/T/fnm_multishells/98961_1715670776511/bin/node
      Yarn: 1.22.22 - /var/folders/md/1qkynwbs1216d57grkbc682h0000gn/T/fnm_multishells/98961_1715670776511/bin/yarn
      npm: 9.8.1 - /var/folders/md/1qkynwbs1216d57grkbc682h0000gn/T/fnm_multishells/98961_1715670776511/bin/npm
      Watchman: 2024.01.22.00 - /opt/homebrew/bin/watchman
    Managers:
      CocoaPods: 1.15.2 - /opt/homebrew/bin/pod
    SDKs:
      iOS SDK:
        Platforms: DriverKit 23.4, iOS 17.4, macOS 14.4, tvOS 17.4, visionOS 1.1, watchOS 10.4
    IDEs:
      Android Studio: 2023.1 AI-231.9392.1.2311.11330709
      Xcode: 15.3/15E204a - /usr/bin/xcodebuild
    npmPackages:
      expo: ~51.0.5 => 51.0.5 
      expo-router: ~3.5.12 => 3.5.12 
      react: 18.2.0 => 18.2.0 
      react-dom: 18.2.0 => 18.2.0 
      react-native: 0.74.1 => 0.74.1 
      react-native-web: ~0.19.10 => 0.19.11 
    npmGlobalPackages:
      eas-cli: 8.0.0
    Expo Workflow: bare

### Expo Doctor Diagnostics

npx expo-doctor@latest
Need to install the following packages:
expo-doctor@1.6.0
Ok to proceed? (y) y
✔ Check Expo config for common issues
✔ Check package.json for common issues
✔ Check dependencies for packages that should not be installed directly
✔ Check npm/ yarn versions
✔ Check for issues with metro config
✖ Check for common project setup issues
✔ Check Expo config (app.json/ app.config.js) schema
✔ Check that native modules do not use incompatible support packages
✔ Check for legacy global CLI installed locally
✔ Check native tooling versions
✔ Check that packages match versions required by installed Expo SDK
✔ Check that native modules use compatible support package versions for installed Expo SDK

Detailed check results:

This project has native project folders but also has config plugins, indicating it is configured to use Prebuild. EAS Build will not sync your native configuration if the ios or android folders are present. Add these folders to your .gitignore file if you intend to use prebuild (aka "managed" workflow).
This project has multiple package manager lock files (yarn.lock, package-lock.json). This may cause EAS build to restore dependencies with a different package manager from what you use in other environments.

One or more checks failed, indicating possible issues with the project.

**Comment by mlecoq** [comment]
Fyi the issue has been introduced in `13.0.5`

**Comment by K-E-L** [comment]
I am still getting this issue on 13.0.9

**Comment by cjoshmarshall** [comment]
![Screenshot 2024-07-30 090511](https://github.com/user-attachments/assets/28df27b6-9b52-46fe-8565-4ff65c8f1713)
![1000069665](https://github.com/user-attachments/assets/88faaf74-1f63-45b8-a79d-beb9e5196b16)

**Comment by behenate** [comment]
@cjoshmarshall Are there any specific steps to reproduce the issue or adding an event listener is enough?

**Comment by cjoshmarshall** [comment]
@behenate Just adding an event listener is enough to break the app. To confirm, I bootstrapped a new app using create-expo-app and only added these few lines of code.

```
useEffect(() => {
    DeviceMotion.addListener(() => {
      console.log("Device Motion");
    });
    return () => {
      DeviceMotion.removeAllListeners();
    };
  }, []);
```

**Comment by adrixgc** [comment]
Any updates on this? It is closed but the bug is still reproducible in 13.0.9

**Comment by behenate** [comment]
@adrixgc 
I will need a more complete example to investigate further. I pasted @cjoshmarshall's code into a fresh SDK 52 project and it seems to be working fine
